src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -358,4 +383,113 @@ TEST_F(RetrieveInstancesWidgetTestStarted, ReloadFails) {
 
   VerifyAllElementsAreEnabled();
 }
+
+TEST_F(RetrieveInstancesWidgetTestStarted, ProjectChangeSuccessful) {
+  // Note: The project combo box ui tests do not work with QTest::mouseClick, since it not possible
+  // to click on a combobox item with the QTest framework. setCurrentIndex is used instead.
+
+  // The combobox content looks like the following (> marks selection):
+  // > "Default Project (Test Project 1)"
+  // "Test Project 1 (default)"
+  // "Test Project 2"
+
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(kTestProject1),
+                            InstanceListScope(InstanceListScope::kOnlyOwnInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(kTestInstancesProject1)));
+
+  project_combo_box_->setCurrentIndex(1);
+  QCoreApplication::processEvents();
+
+  EXPECT_EQ(loading_started_spy_.count(), 1);
```

Comment:
Couldn't this 4 values be checked by some static function? Seems to be widely used.

Reply:
done. I added `VerifyAndClearSignalsOfSuccessfulLoadingCycle()` and  `VerifyAndClearSignalsOfFailedLoadingCycle()`